manifests/server/puppetserver.pp

```diff
@@ -41,13 +41,22 @@
   $jvm_max_heap_size = $::puppet::server_jvm_max_heap_size,
   $jvm_extra_args    = $::puppet::server_jvm_extra_args,
 ) {
+  
+  File {
+    notify => Class['puppet::server::service'],
+  }
+
+  $puppetserver_config_dir = versioncmp($::puppetversion, '4.0.0') >= 0 ? {
```

Comment:
`$puppetserver_config_dir = num2bool(versioncmp($::puppetversion, '4.0.0') >= 0) ? {`

That fixed the test for me.